google/cloud/spanner/row_parser.h

```diff
@@ -97,30 +128,29 @@ class RowParser {
     }
 
     friend bool operator==(iterator const& a, iterator const& b) {
-      return std::tie(a.curr_, a.it_, a.end_) ==
-             std::tie(b.curr_, b.it_, b.end_);
+      // Input iterators may only be compared to themselves and end.
```

Comment:
Perhaps s/themselves/(copies of) themselves/?

Reply:
done.